src/lap_stats/lap_stats.c

```diff
@@ -82,15 +82,210 @@ static float g_distance;
+        GeoPoint gp;
+        struct GeoCircle gc;
+
+        gp = getStartPoint(track);
+        gc = gc_createGeoCircle(gp, radius);
+        g_start_geo_trigger = createGeoTrigger(&gc);
+
+        /*
+         * Trip start geo trigger as unit may get powered up in start
+         * geo circle
+         */
+        geo_trigger_trip(&g_start_geo_trigger);
+
+        gp = getFinishPoint(track);
+        gc = gc_createGeoCircle(gp, radius);
+        g_finish_geo_trigger = createGeoTrigger(&gc);
+}
+
+static void update_sector_geo_circle(const int sector)
+{
+        const GeoPoint point =
+                getSectorGeoPointAtIndex(&g_active_track, sector);
+        g_geo_circles.sector = gc_createGeoCircle(point, g_geo_circle_radius);
+}
+
+/**
+ * Sets the start and finish geo circles used for start and finish lines
+ * respectively.
+ * @param track Pointer to the track where we are at.
+ * @param radius Radius of the geo circles at this track in meters.
+ */
+static void setup_geo_circles(const Track *track, const float radius)
+{
+        const GeoPoint start_p = getStartPoint(track);
+        g_geo_circles.start = gc_createGeoCircle(start_p, radius);
+
+        const GeoPoint finish_p = getFinishPoint(track);
+        g_geo_circles.finish = gc_createGeoCircle(finish_p, radius);
+
+        /* Set sector geo circle to first circle */
+        update_sector_geo_circle(0);
+}
+
+/**
+ * Tells us if we have valid start and finish points for a given track.
+ * @return true if we do, false otherwise.
+ */
+static bool isStartFinishEnabled(const Track *track)
+{
+        return isFinishPointValid(track) && isStartPointValid(track);
+}
+
+/**
+ * Tells us if we have sectors defined for this track.
+ * @return true if we do, false otherwise.
+ */
+static bool isSectorTrackingEnabled(const Track *track)
+{
+        if (!isStartFinishEnabled(track))
+                return 0;
+
+        /*
+         * Must have >= one valid sector; must start at position 0.  Be careful
+         * when using getSectorGeoPointAtIndex because if a point is in within
+         * a valid range but is an invalid GPS value, it will return the finish
+         * line (logical because the last sector boundary is always the finish
+         * line).  So we must compare sector 0 against the finish line and if
+         * they are the same, then we do not have sector values defined.
+         */
+        const GeoPoint sp0 = getSectorGeoPointAtIndex(track, 0);
+        const GeoPoint fin = getFinishPoint(track);
+        return !are_geo_points_equal(&fin, &sp0);
+}
+
+
+/**
+ * Sets the active track that lap_stats uses for determining its various
+ * statistics.
+ * @param track Pointer to the track where we are at.
+ * @param radius Radius of the geo circles at this track in meters.
+ * @param track_status Status indicating who set the track value.
+ * @return true if setup was successful, false otherwise.
+ */
+static bool _set_active_track(const Track *track, const float radius,
```

Comment:
I tend to use the _<method name> prefix to indicate that the caller should really understand what they are doing when they invoke this method.  Its a practice I learned when doing Linux work and is one that I am still refining admittedly.  In this case its because both `lapstats_set_active_track` and `lapstats_setup` both invoke this method as it is common code to them and to no-one else.  I suppose I could drop the leading `_` and just make it set_active_track, but then that might get confusing with `lapstats_set_active_track`